### PR TITLE
Issue #514 : Fix intermittent issue with autocomplete on definition search

### LINKED ISSUE #514: Autocomplete on definition search sometimes doesn't
Every so often if I remove a character from a definition i'm searching for or just type in the definition I'll get a "no matches found" even though I'm sure I've seen that component before. This happens intermittently.

### PR DESCRIPTION
**Pull request by storrisi**
Fixes #514 

The issue is fixed by upgrading `react-bootstrap-typeahead`. To reproduce the issue more reliably in the current version you can slow down the network speed to "Slow 3G" in Chrome Dev Tools

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     "pako": "^1.0.6",
     "react": "^16.4.1",
     "react-bootstrap": "^0.32.3",
-    "react-bootstrap-typeahead": "^2.6.0",
+    "react-bootstrap-typeahead": "^3.3.0",
```
Comment (dabutvin) [inline package.json]: very nice! 

Seems to be greatly improved

**Comment by dabutvin** [comment]
I am getting consistent e2e failures with this change.
I think the definitions map needs to be adjusted for the new markup

old markup

```html
<div class="show-grid spacer row">
  <div class="col-md-10 col-md-offset-1">
    <div class="rbt open clearfix" tabindex="-1" style="position: relative;">
      <div class="rbt-input form-control" tabindex="-1">
        <div class="rbt-input-wrapper">
        <div style="display: block; position: relative;">
        <input aria-activedescendant="" aria-autocomplete="both" aria-expanded="false" aria-haspopup="listbox" aria-owns="rbt-menu-t65g4wvf1gq" autocomplete="off" role="combobox" placeholder="Component search..." type="text" class="rbt-input-main" value="" style="background-color: transparent; border: 0px; box-shadow: none; cursor: inherit; outline: none; padding: 0px; width: 100%;">
        <div style="bottom: 0px; display: block; pointer-events: none; position: absolute; top: 0px; z-index: 0;">
          <input aria-hidden="true" tabindex="-1" class="rbt-input-hint" value="" style="background-color: transparent; border: 0px; box-shadow: none; cursor: inherit; outline: none; padding: 0px; color: rgba(0, 0, 0, 0.35); box-sizing: content-box; width: 1px;">
          <div style="height: 0px; left: 0px; overflow: scroll; position: absolute; top: 0px; visibility: hidden; white-space: pre; font-size: 14px; font-family: Montserrat, sans-serif; font-weight: 400; font-style: normal; letter-spacing: normal; text-transform: none;"></div>
      </div>
      </div>
      </div>
    </div>
    <div aria-atomic="true" aria-live="polite" class="sr-only rbt-sr-status" role="status">0 selections</div>
    </div>
  </div>
</div>
```

new markup

```html
<div class="show-grid spacer row">
  <div class="col-md-10 col-md-offset-1">
    <div class="rbt clearfix open" tabindex="-1" style="position: relative;">
      <div class="rbt-input-hint-container" style="position: relative;">
        <input aria-activedescendant="" aria-autocomplete="both" aria-expanded="false" aria-haspopup="listbox" aria-owns="" autocomplete="nope" placeholder="Component search..." role="combobox" type="text" class="rbt-input-main form-control rbt-input " value="">
        <div style="left: 0px; pointer-events: none; position: absolute; top: 0px; display: inline-block;">
          <input aria-hidden="true" readonly="" tabindex="-1" class="rbt-input-hint" value="" style="background-color: transparent; border-color: transparent; box-shadow: none; color: rgba(0, 0, 0, 0.35); box-sizing: content-box; width: 3px; border-style: solid; border-width: 1px; font-size: 14px; line-height: 20px; margin: 0px; padding: 6px 12px;">
          <div style="height: 0px; left: 0px; overflow: scroll; position: absolute; top: 0px; visibility: hidden; white-space: pre; font-size: 14px; font-family: Montserrat, sans-serif; font-weight: 400; font-style: normal; letter-spacing: normal; text-transform: none;"></div>
        </div>
      </div>
      <div aria-atomic="true" aria-live="polite" class="sr-only rbt-sr-status" role="status">0 selections</div>
    </div>
  </div>
</div>
```

**Comment by dabutvin** [comment]
there seems to be an issue on the harvest page here.
Not reproing in dev or prod, just this branch

go to /harvest and pick a component to demo it


There is another issue more specific to github and maven
we are using a method that was removed in this version
`this._typeahead._updateText is not a function`

**Comment by gianpaj** [comment]
I've downgraded back the version of `react-bootstrap-typeahead` but disabled the cache

**Comment by dabutvin** [comment]
👍 this works. We are using browser cache on these api calls anyways, so they read from disk instead of network

**Pull request merged**